BraveRewardsUI/Wallet/RewardsSummaryProtocol.swift

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+/// Shared resources for showing summary of all BAT rewards.
+protocol RewardsSummaryProtocol {
+  var state: RewardsState { get }
+  
+  /// Month and year of which the rewards summary is shown.
+  var summaryPeriod: String { get }
+  
+  /// Rows showing different types of earnings, tips etc.
+  var summaryRows: [RowView] { get }
+  
+  /// A view informing users about contributing to unverified publishers.
+  var disclaimerView: DisclaimerView { get }
```

Comment:
If we're going to put this in the protocol it has to be optional so it can be hidden when the user doesn't have auto-contribute enabled or doesn't have any auto-contribution entries.  Maybe we should get clarification from @NejcZdovc or just double check in brave-core when that disclaimer is supposed to be shown.